src/core/frames/frame_controller.ts

```diff
@@ -295,3 +295,12 @@ function activateElement(element: Node | null) {
     return element
   }
 }
+
+function elementIsNavigable(element: Element) {
```

Comment:
For the sake of passing the suite, this function is directly copied from the implementation in `Session`.

It's unclear to me the difference between `FormSubmitObserver` (and its `Session` delegate) and `FormInterceptor` (and its `FrameController` delegate). Is the concept of a page-level delegate dissimilar enough from a frame-level delegate that the two concepts should be distinctly represented with their own interfaces?

Would it be worthwhile to re-use the `FormSubmitObserver` interface in the case of a `Session` and `FrameController`, then leave it up to the delegate implementations to have different behaviors?

There feels like there is a concept we're missing here, which is leading to the duplicated guard logic.

Reply:
> For the sake of passing the suite, this function is directly copied from the implementation in Session.

Let's demote `Session#elementIsNavigable()` to a function, export it from `session`, and import it here.

> Is the concept of a page-level delegate dissimilar enough from a frame-level delegate that the two concepts should be distinctly represented with their own interfaces?

I'm pretty sure the duplication is an artifact of the original \<turbolinks-frame> implementation in HEY. Please do explore unification, if you're so inclined (in a separate branch 😄)!

Reply:
@sstephenson I've opened https://github.com/hotwired/turbo/pull/232 to remove `FormInterceptor` and use `FormSubmitObserver` universally.